src/main/java/uk/gov/hmcts/ccd/domain/model/casedataaccesscontrol/RoleAssignments.java

```diff
@@ -14,21 +18,44 @@
 @Builder
 @Data
 @AllArgsConstructor
-@NoArgsConstructor
 public class RoleAssignments {
     private List<RoleAssignment> roleAssignments;
 
+    private static final String ORGANISATION = "Organisation";
+    private final CaseDefinitionRepository caseDefinitionRepository;
+
+    @Autowired
+    public RoleAssignments(@Qualifier(CachedCaseDefinitionRepository.QUALIFIER)
+                               CaseDefinitionRepository caseDefinitionRepository) {
+        this.caseDefinitionRepository = caseDefinitionRepository;
+    }
+
     @JsonIgnore
     public List<String> getJurisdictions() {
-        return roleAssignments.stream()
-            .map(RoleAssignment::getAttributes)
-            .filter(Objects::nonNull)
-            .map(RoleAssignmentAttributes::getJurisdiction)
-            .filter(Objects::nonNull)
-            .filter(Optional::isPresent)
-            .map(Optional::get)
-            .distinct()
-            .collect(Collectors.toList());
+        List<String> jurisdictions = new ArrayList<>();
+
+        roleAssignments.stream()
+            .filter(roleAssignment -> !roleAssignment.getGrantType().equals(GrantType.BASIC.name()))
+            .collect(Collectors.toList())
+            .forEach(roleAssignmentAttributes -> {
+                if (roleAssignmentAttributes.getAttributes() != null) {
+                    if (Objects.nonNull(roleAssignmentAttributes.getAttributes().getJurisdiction())) {
```

Comment:
```suggestion
                    if (attributes.getJurisdiction().isPresent()) {
```
Might be better to use this on an `Optional` (same comment applies to others)

Reply:
when the class is initialized from the row in the database, if the values are not in the db it is returned as null, then the isPresent() throws a nullPointer